src/addresses.py

```diff
@@ -1,281 +1,34 @@
-
-    data = unhexlify(hexdata)
-    checksum = data[-4:]
-
-    sha = hashlib.new('sha512')
-    sha.update(data[:-4])
-    currentHash = sha.digest()
-    sha = hashlib.new('sha512')
-    sha.update(currentHash)
-
-    if checksum != sha.digest()[0:4]:
-        status = 'checksumfailed'
-        return status, 0, 0, ''
-
-    try:
-        addressVersionNumber, bytesUsedByVersionNumber = decodeVarint(data[:9])
-    except varintDecodeError as e:
-        logger.error(str(e))
-        status = 'varintmalformed'
-        return status, 0, 0, ''
-
-    if addressVersionNumber > 4:
-        logger.error('cannot decode address version numbers this high')
-        status = 'versiontoohigh'
-        return status, 0, 0, ''
-    elif addressVersionNumber == 0:
-        logger.error('cannot decode address version numbers of zero.')
-        status = 'versiontoohigh'
-        return status, 0, 0, ''
-
-    try:
-        streamNumber, bytesUsedByStreamNumber = \
-            decodeVarint(data[bytesUsedByVersionNumber:])
-    except varintDecodeError as e:
-        logger.error(str(e))
-        status = 'varintmalformed'
-        return status, 0, 0, ''
-
-    status = 'success'
-    if addressVersionNumber == 1:
-        return status, addressVersionNumber, streamNumber, data[-24:-4]
-    elif addressVersionNumber == 2 or addressVersionNumber == 3:
-        embeddedRipeData = \
-            data[bytesUsedByVersionNumber + bytesUsedByStreamNumber:-4]
-        if len(embeddedRipeData) == 19:
-            return status, addressVersionNumber, streamNumber, \
-                b'\x00' + embeddedRipeData
-        elif len(embeddedRipeData) == 20:
-            return status, addressVersionNumber, streamNumber, \
-                embeddedRipeData
-        elif len(embeddedRipeData) == 18:
-            return status, addressVersionNumber, streamNumber, \
-                b'\x00\x00' + embeddedRipeData
-        elif len(embeddedRipeData) < 18:
-            return 'ripetooshort', 0, 0, ''
-        elif len(embeddedRipeData) > 20:
-            return 'ripetoolong', 0, 0, ''
-        return 'otherproblem', 0, 0, ''
-    elif addressVersionNumber == 4:
-        embeddedRipeData = \
-            data[bytesUsedByVersionNumber + bytesUsedByStreamNumber:-4]
-        if embeddedRipeData[0:1] == b'\x00':
-            # In order to enforce address non-malleability, encoded
-            # RIPE data must have NULL bytes removed from the front
-            return 'encodingproblem', 0, 0, ''
-        elif len(embeddedRipeData) > 20:
-            return 'ripetoolong', 0, 0, ''
-        elif len(embeddedRipeData) < 4:
-            return 'ripetooshort', 0, 0, ''
-        x00string = b'\x00' * (20 - len(embeddedRipeData))
-        return status, addressVersionNumber, streamNumber, \
-            x00string + embeddedRipeData
-
-
-def addBMIfNotPresent(address):
-    """Prepend BM- to an address if it doesn't already have it"""
-    address = str(address).strip()
```

Comment:
LOL, you've removed an addresses manipulation part of the protocol implementation (;